signature/mechanism.go

```diff
@@ -13,6 +13,10 @@ import (
 // FIXME: Eventually expand on keyIdentity (namespace them between mechanisms to
 // eliminate ambiguities, support CA signatures and perhaps other key properties)
 type SigningMechanism interface {
+	// ImportKeysFromBlob imports public keys from the supplied blob and returns their identities.
+	// The blob is assumed to have an appropriate format (the caller is expected to know which one).
+	// NOTE: This may modify long-term state (e.g. key storage in a directory underlying the mechanism).
+	ImportKeysFromBlob(blob []byte) ([]string, error)
```

Comment:
much as you did here https://github.com/projectatomic/skopeo/pull/55/files#diff-230170c6339d5a6b9de551a0aeb6e74eR44